### ISSUE TITLE
Cant upgrade an application when the current version is no longer in the helm repository

### ISSUE THREAD

**Issue by hatharom** [rancher/rancher#27224]
Rancher | v2.4.3
User Interface | v2.4.22

Say we have a deployed app with chart version of 1.1

The helm repositry maintainer announces a new version as 1.2 but deletes the old one. From mistake or intentionally.. doesnt matter.

The App page says ` Upgrade available(1.2)`
But when I click on upgrade nothing happens. 
If we put back the old version it will work again.

why??
Rancher doesnt log anything.

pretty annoying issue and totally unacceptable. Because of this rancher is dangerous to use.
For any reason at any time you wont be able to upgrade anymore ever.

edit: It should be supported because helm supports it also, if I change my helm chart , increment chart version (the old version is deployed but as chart source long gone) then I am able to run` helm upgrade`

**Comment by mschiessl** [comment]
+1
As the same thing happens to me while even having the older version availaible in my chart repo.

Installed App: version: 1.0
Upgradeable Version: 1.1

Clicking on Upgrade - grays out the background for a second and returns then to the apps screen.

rancher:  | v2.4.3
User Interface | v2.4.22

looking into the logs i see the following:
```
2020-05-23 12:56:45.014293 W | etcdserver: read-only range request "key:\"/registry/management.cattle.io/catalogtemplates/\" range_end:\"/registry/management.cattle.io/catalogtemplates0\" " with result "range_response_count:404 size:539263" took too long (214.986259ms) to execute
```
actually this is the only logline i was able to set into correlation with the issue.

**Comment by mschiessl** [comment]
@hatharom -
Can you please check if the Chart version follows the SemVer guidance (https://semver.org/)
I just movey my chart name from 1.0 to 1.0.0 and it seems to work again.

As the definition of Charts.yaml i found my version did not match the SemVer and so i updated it and now it'S working fine. Please give it a shot

**Comment by hatharom** [comment]
Yes they are valid. The deployed version is 1.1.0 the new one is 1.2.0

**Comment by mschiessl** [comment]
ok - for me it is actually working since i've changed the chart versioning

**Comment by ChrML** [comment]
Same issue here in Rancher 2.4.3.

We have done multiple upgrades after upgrading to Rancher 2.4.3, but somehow it became an issue today. Clicking the "Upgrade" button just blanks the page for 1 second, then goes back to the "Apps" page. Suddenly none of our workloads can be upgraded anymore ...

**Comment by ChrML** [comment]
Possible duplicate of #27025.

**Comment by ChrML** [comment]
As an additional data point: We just downgraded Rancher from 2.4.3 to 2.3.7 and this is no longer an issue. Now we are able to upgrade apps again like before.

**Comment by mschiessl** [comment]
@ChrML, have a look at the dup mentioned by you - i think the root cause is explained well in there ;)

**Comment by ChrML** [comment]
@mschiessl Thanks! I didn't spend much more time investigating, but likely one of two causes:

*  Current version was deleted. (however in our case it was not)

*  Not correct version format. Our version numbers look like 20.5.24.1234 and tick upward. Did not try to change it as never been a problem before on Rancher 2.3.x.

What is strange is that for us, we could upgrade no applications at all. We have about 20 different types of applications with different charts, and none could be upgraded. However it was possible to launch new apps.

**Comment by pahansen95** [comment]
After experiencing the same issue (Rancher v2.4.3 & UI v2.4.22) after updating a private Helm Chart I was able to force an upgrade by using the API. It also subsequently fixed the UI bug. I have listed my steps below:

  * You can submit the API request in the UI by selecting the kebob/more options button & selecting the `View in API` option.
  * It will take you to a text page w/ a box in the upper right hand corner with an `upgrade` button.
  * Before hitting the upgrade button I copied the value of the `externalId` key found in the text on the page.
  * After hitting the upgrade button I pasted in the value I just copied into the `externalId` text box & then updated the version number to reflect the newest version.
    * **before**: `catalog://?catalog=my-catalog&template=my-app&version=0.1.0`
    * **after**: `catalog://?catalog=my-catalog&template=my-app&version=0.1.1`
 * I then submitted the request & my app upgraded as expected.

I imagine you could also update your answers or the actual HELM values using the `answers` & `valuesYaml` in the API request but I haven't tried.

**Comment by ChrML** [comment]
This was resolved for me by changing version number semantics from 20.5.24.1234 to 20.24.1234.
